### PR TITLE
feat: dynamic shape support for pow/mod/eq operator

### PR DESCRIPTION
**Pull request by keehyuna**
# Description

dynamic shape support for pow/mod/eq operator
Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 1]
```diff
@@ -2278,6 +2279,7 @@ def aten_ops_bitwise_not(
 
 @dynamo_tensorrt_converter(torch.ops.aten.eq.Tensor)
 @dynamo_tensorrt_converter(torch.ops.aten.eq.Scalar)
+@dynamo_tensorrt_converter(operator.eq, supports_dynamic_shapes=True)
```
Comment (keehyuna) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2282]: I will rebase after https://github.com/pytorch/TensorRT/pull/2979/files#diff-9cbf43cfb62cdf682eef77f6fd5cabc488bb5a91ff09bfee5d593827542b2476R2256-R2257 is merged

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 2]
```diff
@@ -2002,6 +2002,7 @@ def aten_ops_div(
 @dynamo_tensorrt_converter(
     torch.ops.aten.pow.Tensor_Scalar, supports_dynamic_shapes=True
 )
+@dynamo_tensorrt_converter(operator.pow, supports_dynamic_shapes=True)
```
Comment (chohk88) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2005]: From the PR (https://github.com/pytorch/TensorRT/pull/2918), it looks like dynamic testing and registration for `operator.mul` were added because PyTorch internally uses `operator.mul` when using `torch.nn.Linear` layers. This might be why `operator.mul` was registered with the converter. 

If I'm misunderstanding, could you please explain? (cc. @peri044 )

Then, in this PR, why do we need to register `operator.pow`, `operator.eq`, and `operator.mod` with the converter?

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2005]: It seems `operator.*` overrides original Python ops, which results in running any Python op will create a TRT Layer to handle it. However, they are not listed in [the schema](https://github.com/pytorch/pytorch/blob/a408920817ac3852d27fc96ba85109ff4c981d52/aten/src/ATen/native/native_functions.yaml). I'm also curious if these ops' converters are needed

Comment (keehyuna) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2005]: Thanks @chohk88 / @zewenli98
If operator.pow is missing and ** is used in module, there is exception from TRTInterpreter.

E torch_tensorrt.dynamo.conversion._TRTInterpreter.UnsupportedOperatorException: Conversion of function _operator.pow not currently supported!

I saw such usage of operator in openchat model. I think we need to register these operators to support such usage.
https://github.com/imoneoi/openchat/blob/master/ochat/models/unpadded_llama.py#L105

Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2005]: Thanks for the example. If then, it seems to be necessary I think. 
cc: @narendasan @dheerajperi

File: tests/py/dynamo/conversion/test_pow_aten.py [hunk 3]
```diff
@@ -59,6 +59,65 @@ def forward(self, lhs_val):
...
+                torch.float,
+            ),
+        ]
+    )
+    def test_pow_dynamic_shape(
+        self, _, min_shape, opt_shape, max_shape, type, output_type
+    ):
+        class pow(nn.Module):
+            def forward(self, lhs_val, rhs_val):
+                return torch.ops.aten.floor_divide.default(lhs_val, rhs_val)
```
Comment (peri044) [inline tests/py/dynamo/conversion/test_pow_aten.py:87]: Why does pow test case use floor_divide here ?

**Review by peri044 — COMMENTED** [review]
Overall LGTM. Added a question

File: tests/py/dynamo/conversion/test_pow_aten.py (hunk 3 above)
Comment (keehyuna) [reply tests/py/dynamo/conversion/test_pow_aten.py:87]: Thanks for pointing this. I fixed.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 2 above)
Comment (chohk88) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:2005]: Thanks for the example. LGTM!
